mesonbuild/compilers/detect.py

```diff
@@ -572,6 +577,36 @@ def sanitize(p: str) -> str:
             return cls(
                 ccache, compiler, version, for_machine, is_cross, info,
                 exe_wrap, full_version=full_version, linker=linker)
+        
+        if 'Metrowerks C/C++' in out or 'Freescale C/C++' in out:
+            if 'ARM' in out:
+                cls = c.MetrowerksCCompilerARM if lang == 'c' else cpp.MetrowerksCPPCompilerARM
+                lnk = linkers.MetrowerksLinkerARM
+            else:
+                cls = c.MetrowerksCCompilerEmbeddedPowerPC if lang == 'c' else cpp.MetrowerksCPPCompilerEmbeddedPowerPC
+                lnk = linkers.MetrowerksLinkerEmbeddedPowerPC
+
+            mwcc_ver_match = re.search(r'Version (\d+)\.(\d+)\.?(\d+)? build (\d+)', out)
+            assert mwcc_ver_match is not None, 'for mypy'  # because mypy *should* be complaning that this could be None
+            compiler_version = '.'.join([x for x in mwcc_ver_match.groups() if x is not None])
+
+            env.coredata.add_lang_args(cls.language, cls, for_machine, env)
+            ld = env.lookup_binary_entry(for_machine, cls.language + '_ld')
+
+            if ld is not None:
+                _, o_ld, _ = Popen_safe(ld + ['--version'])
+
+                mwld_ver_match = re.search(r'Version (\d+)\.(\d+)\.?(\d+)? build (\d+)', o_ld)
+                assert mwld_ver_match is not None, 'for mypy'  # because mypy *should* be complaning that this could be None
+                linker_version = '.'.join([x for x in mwld_ver_match.groups() if x is not None])
```

Comment:
```suggestion
                linker_version = '.'.join(x for x in mwld_ver_match.groups() if x is not None)
```

You dont need an intermediate list. Just a generative expression. `filter()` would work as well, but I'm not sure everybody like it...